library/systemd/src/lib/yast2/systemd_unit.rb

```diff
@@ -121,11 +129,13 @@ def initialize(full_unit_name, propmap = {}, property_text = nil)
       @name = id.to_s.split(".").first || unit_name
     end
 
-    # @raise [Yast::SystemctlError] if 'systemctl show' cannot be executed
+    # @raise [Yast::CouldNotRefreshUnitError] if 'systemctl show' cannot be executed
     def refresh!
       @properties = show
       @error = properties.error
       properties
+    rescue Yast::SystemctlError
+      raise Yast::CouldNotRefreshUnitError, self
```

Comment:
shouldn't it set also `@error` when show failed? Otherwise I worry we will be in an inconsistent state.

Reply:
Good catch. 

I am not sure about what to do.

@joseivanlopez, thoughts?

Reply:
Something like this? `@error = "Fails to refresh unit"` ?

Reply:
Okey, thank you